sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -0,0 +1,181 @@
+    ///   - credential: The user credential.
+    ///   - options: Options used to configure the client.
+    public init(
+        endpoint: String,
+        credential: CommunicationUserCredential,
+        withOptions options: AzureCommunicationChatClientOptions
+    ) throws {
+        self.endpoint = endpoint
+        self.credential = credential
+        self.options = options
+
+        guard let baseUrl = URL(string: endpoint) else {
+            throw AzureError.client("Unable to form base URL.")
+        }
+
+        let authPolicy = CommunicationUserCredentialPolicy(credential: credential)
+
+        self.client = try AzureCommunicationChatClient(baseUrl: baseUrl, authPolicy: authPolicy, withOptions: options)
+    }
+
+    // MARK: Public Methods
+
+    /// Get a ChatThreadClient.
+    /// - Parameters:
+    ///   - threadId: The threadId.
+    public func getChatThreadClient(threadId: String) throws -> ChatThreadClient {
+        return try ChatThreadClient(
+            threadId: threadId,
+            endpoint: endpoint,
+            credential: credential,
+            withOptions: options
+        )
+    }
+
+    /// Create a ChatThread.
+    /// - Parameters:
+    ///   - request: Request for creating a chat thread with the topic and members to add.
+    ///   - withOptions: Create chat thread options.
+    ///   - completionHandler: A completion handler that receives a ChatThreadClient on success.
+    public func createChatThread(
+        request: CreateChatThreadRequest,
+        withOptions options: CreateChatThreadOptions? = nil,
+        completionHandler: @escaping (Result<ChatThreadClient, Error>) -> Void
+    ) {
+        client.create(chatThread: request, withOptions: options) { result, _ in
+            do {
+                switch result {
+                case let .success(createThreadResponse):
+                    var responseId: String?
+                    for response in createThreadResponse.multipleStatus ?? [] {
+                        if response.id?.hasSuffix("@thread.v2") ?? false,
+                           response.type ?? "" == "Thread" {
+                            responseId = response.id
+                        }
+                    }
+
+                    guard let threadId = responseId else {
+                        throw AzureError.service("Service response does not contain ThreadId.")
+                    }
+
+                    let chatThreadClient = try ChatThreadClient(
+                        threadId: threadId,
+                        endpoint: self.endpoint,
+                        credential: self.credential,
+                        withOptions: self.options
+                    )
+                    completionHandler(.success(chatThreadClient))
+
+                case let .failure(error):
+                    throw error
+                }
+            } catch {
+                completionHandler(.failure(error))
+            }
```

Comment:
Would recommend some comments here to describe this behavior. It is atypical to NOT return whatever the service call returned.